### PR TITLE
Refactor EntryEditor & EntryEditorTab

### PR DESCRIPTION
**Pull request by bartsch-dev**
I refactored many comments (some of them even referred to variables which do not exist anymore).
I removed some unused Methods, replaced some code with already existing methods and refactored a switch-case.

### CODE REVIEW

File: src/main/java/net/sf/jabref/gui/entryeditor/EntryEditor.java [hunk 1]
```diff
@@ -328,7 +327,8 @@ private void addSourceTab() {
         srcPanel.setName(panelName);
         tabbed.addTab(panelName, IconTheme.JabRefIcon.SOURCE.getSmallIcon(), srcPanel, toolTip);
         tabs.add(srcPanel);
-        sourceIndex = tabs.size() - 1; // Set the sourceIndex variable.
+        // Set the sourceIndex variable
```
Comment (tschechlovdev) [inline src/main/java/net/sf/jabref/gui/entryeditor/EntryEditor.java]: I think the comment can be deleted.

File: src/main/java/net/sf/jabref/gui/entryeditor/EntryEditorTab.java [hunk 2]
```diff
@@ -300,9 +281,7 @@ public FieldEditor getActive() {
 
     public void activate() {
         if (activeField != null) {
-            /**
-             * Corrected to fix [ 1594169 ] Entry editor: navigation between panels
-             */
+            // Corrected to fix [ 1594169 ] Entry editor: navigation between panels
```
Comment (tschechlovdev) [inline src/main/java/net/sf/jabref/gui/entryeditor/EntryEditorTab.java]: This can be also deleted I think. Or you could replace it to not point to the bug.

File: src/main/java/net/sf/jabref/gui/entryeditor/EntryEditor.java (hunk 1 above)
Comment (bartsch-dev) [inline src/main/java/net/sf/jabref/gui/entryeditor/EntryEditor.java]: done

File: src/main/java/net/sf/jabref/gui/entryeditor/EntryEditorTab.java (hunk 2 above)
Comment (bartsch-dev) [inline src/main/java/net/sf/jabref/gui/entryeditor/EntryEditorTab.java]: done

**Comment by grimes2** [comment]
In the prompt Firstname Lastname... "and others" is not translated. My proposal is to replace it by "et al.". This is international, so no translation needed.
`prompt = String.format("%1$s and %1$s and others", Localization.lang("Firstname Lastname"))`

File: src/main/java/net/sf/jabref/gui/entryeditor/EntryEditor.java [hunk 3]
```diff
@@ -231,19 +236,13 @@ private void setupFieldPanels() {
...
         Set<String> deprecatedFields = new HashSet<>(EntryConverter.FIELD_ALIASES_TEX_TO_LTX.keySet());
         Set<String> usedOptionalFieldsDeprecated = new HashSet<>(deprecatedFields);
 
         if ((type.getOptionalFields() != null) && !type.getOptionalFields().isEmpty()) {
             if (!frame.getCurrentBasePanel().getBibDatabaseContext().isBiblatexMode()) {
-                displayedOptionalFields.addAll(type.getOptionalFields());
-
                 addOptionalTab(type);
             } else {
-                displayedOptionalFields.addAll(type.getPrimaryOptionalFields());
```
Comment (stefan-kolb) [inline src/main/java/net/sf/jabref/gui/entryeditor/EntryEditor.java]: Why is this removed here? If I remember this right this was needed to distinguish optional fields from fields that are displayed in the other fields tab?

**Comment by koppor** [comment]
`and others` is the bibtex string to procude `et al.` using bibtex/biblatex Therefore, this *must not* be translated. It is contained in the help to educate users to use this string.

File: src/main/java/net/sf/jabref/gui/entryeditor/EntryEditor.java (hunk 3 above)
Comment (bartsch-dev) [inline src/main/java/net/sf/jabref/gui/entryeditor/EntryEditor.java]: Items of the list are never used, only added to it (`displayedOptionalFields` was only used in line 234, 240, 244, and 245; or am I missing something?)

**Comment by bartsch-dev** [comment]
Changed the prompt for `Month`-field.
I don't know about the prompts for the other fields, maybe you can point me into the right direction?

**Comment by grimes2** [comment]
From my side it looks ok now. I misinterpreted the author... prompt, because I didn't know the function of `and others` in BibTeX author/editor field. I found this documented in the official specification "Bibtexing".

**Comment by grimes2** [comment]
Just one small addition. I found following person name fields for BibLaTeX:
```
private static final List<String> BIBLATEX_PERSON_NAME_FIELDS = Arrays.asList(FieldName.AUTHOR, FieldName.EDITOR,
            FieldName.EDITORA, FieldName.EDITORB, FieldName.EDITORC, FieldName.TRANSLATOR, FieldName.ANNOTATOR,
            FieldName.COMMENTATOR, FieldName.INTRODUCTION, FieldName.FOREWORD, FieldName.AFTERWORD,
            FieldName.BOOKAUTHOR, FieldName.HOLDER, FieldName.SHORTAUTHOR, FieldName.SHORTEDITOR, FieldName.SORTNAME,
            FieldName.NAMEADDON, FieldName.ASSIGNEE);
```
 I assume that the suggestion syntax
`return String.format("%1$s and %1$s and others", Localization.lang("Firstname Lastname"));`
 is valid for all the mentioned fields above. What do you think? Should we add the missing person name fields to the switch construction?

**Comment by bartsch-dev** [comment]
No I reuse the already existing lists.

**Comment by koppor** [comment]
This PR breaks <kbd>Ctrl</kbd> + <kbd>A</kbd>. See https://github.com/JabRef/jabref/issues/2615
